### DISCUSSION TITLE
Create vcpkg custom port, based of given .dll .h and .lib

### DISCUSSION THREAD

**Discussion by le1nax** [microsoft/vcpkg#34543] [Q&A] [answered]
Hello,

When calling `vcpkg_cmake_install` in the `portfile.make` of my custom port, it tells me the installation failed.
I'm trying to install a simple github repo, that I quickly made up in order to use 'vcpkg_from_github`. This repo only holds the binaries and includes of the library I was provided (.dll .h and .lib). (https://github.com/le1nax/medit-libs)

Now I assume the `CMakeLists.txt` I've arbitrarily written does not properly install these 3 files. Right now it looks like this:
```
cmake_minimum_required (VERSION 3.17)
project(medit-libs CXX)  

add_library(pcanbasic STATIC IMPORTED)
set_target_properties(pcanbasic PROPERTIES
         "${CMAKE_CURRENT_LIST_DIR}/lib/PCANBasic.lib"
)

target_include_directories(pcanbasic INTERFACE "{CMAKE_CURRENT_LIST_DIR}/include")
```

Does anyone know how to properly setup a CMakeLists to install the .dll, .h and .lib in order to make `vcpkg_cmake_install` work?

**Comment by Osyotr** [comment]
You don't really need to write `CMakeLists.txt` for prebuilt libraries. See https://github.com/microsoft/vcpkg/discussions/31195#discussioncomment-5775219

**Reply by le1nax** [reply]
oh I see, so `vcpkg_cmake_install` isnt needed then. What I'm confused with is, that when integrating the package into my cpp cmake managed project by doing `find_package` and afterwards `target_include_directories` and `target_link_libraries`, the build process fails because x86_64-w64-mingw32/bin/ld.exe: cannot find -lpcanbasic: No such file or directory. I assume at some point I'm missing an install. Is there anything left which I'm missing?

**Reply by Osyotr** [reply, marked as answer]
You need to write something similar to this: https://gist.github.com/Osyotr/709e71a91a3b277b01b684a0d638e758

**Reply by le1nax** [reply]
From my understanding a `Find<package>.cmake` is only required for cmake to locate my package in order to make `find_package` work. In my case I used a `<package>Config.cmake` file that does the same job (I think). 
So my `CMakeLists.txt` did accept `find_package(<package> CONFIG REQUIRED)`, but then couldnt do `target_link_libraries` because the `.lib` could not be located. 
Its true that finding the cause of this is probably not possible without me providing more information. I'm also not sure what particular information I should share because the problem could be everywhere I feel like.

**Reply by Osyotr** [reply]
IIUC you're generating cmake config from this CMakeLists.txt: https://github.com/le1nax/medit-libs/blob/master/CMakeLists.txt
I'm not sure that works for imported target. You may want to write cmake config manually (take sqlite3 config as an example).
